src/index.spec.ts

```diff
@@ -1,12 +1,227 @@
+    expect(removeDuplicatesFromArray([1, 'a', 3])).toEqual([1, 'a', 3]);
+  });
+  it('should return [1,2,3] with [1,2,3,2]', () => {
+    expect(removeDuplicatesFromArray([1, 2, 3, 2])).toEqual([1, 2, 3]);
+  });
+  it('should return Error with any type diferent to string[] or number[]', () => {
+    const testFunction = () => {
+      removeDuplicatesFromArray({} as any);
+    };
+    expect(testFunction).toThrow(Error);
+  });
+});
+
+describe('createProduct test', () => {
+  it('should return a product with a objetc verified with joi', () => {
+    const product1 = {
+      name: 'Browneggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs',
+      price: 28.1,
+    };
+    const resp = {
+      id: expect.any(Number),
+      name: 'Browneggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs',
+      price: 28.1,
+    };
+    expect(createProduct(product1)).toEqual(resp);
+  });
+  it('should return Error when object doesnt verified with joi', () => {
+    const product2 = {
+      name: 'Brown eggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs',
+      price: 28.1,
+    };
+    const product3 = {
+      name: 'Brown eggs',
+      tags: {} as any,
+      description: 'Raw organic brown eggs',
+      price: 28.1,
+    };
+    const product4 = {
+      name: 'Brown eggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs this is extra text for this field',
+      price: 28.1,
+    };
+    const product5 = {
+      name: 'Brown eggs',
+      tags: ['dairy'],
+      description: 'Raw organic brown eggs',
+      price: 1,
+    };
+    const testFunction2 = () => {
+      createProduct(product2);
+    };
+    const testFunction3 = () => {
+      createProduct(product3);
+    };
+    const testFunction4 = () => {
+      createProduct(product4);
+    };
+    const testFunction5 = () => {
+      createProduct(product5);
+    };
+    expect(testFunction2).toThrow(Error);
+    expect(testFunction3).toThrow(Error);
+    expect(testFunction4).toThrow(Error);
+    expect(testFunction5).toThrow(Error);
+  });
+});
+
+describe('createRandomProduct test', () => {
+  it('should return a product with correct email', () => {
+    expect(createRandomProduct('clark@kent.com')).toMatchObject({
+      id: expect.anything(),
+      name: expect.anything(),
+      description: expect.anything(),
+      price: expect.anything(),
+      tags: expect.anything(),
+    });
+  });
+  it('should return Error with a email without create rol', () => {
+    const createWithAdmin = () => {
+      createRandomProduct('bruce@wayne.com');
+    };
+    const createWithUser = () => {
+      createRandomProduct('diana@themyscira.com');
+    };
+    const createWithEmpty = () => {
+      createRandomProduct('');
+    };
+    expect(createWithAdmin).toThrow(Error);
+    expect(createWithUser).toThrow(Error);
+    expect(createWithEmpty).toThrow(Error);
```

Comment:
This should be on different `it()` statements, with descriptive test names and error messages.

Reply:
You could have added a test in case the email is not found in the users array